nntrainer/src/tensor.cpp

```diff
@@ -91,6 +92,22 @@ Tensor::Tensor(int batch, int channel, int height, int width) {
   setZero();
 }
 
+bool Tensor::operator== (const Tensor &rhs) const {
+  if (this->dim != rhs.dim)
+    return false;
+
+  if (data.size() != rhs.data.size())
+    return false;
+
+  for (size_t i = 0; i < data.size(); ++i) {
+    if (std::isnan(data[i]) || std::isnan(rhs.data[i]) ||
```

Comment:
What if `data[i]` and `rhs.data[i]` is both `nan`? Would this be considered not eq?

Reply:
https://stackoverflow.com/a/570694
in general, `nan != nan`

Reply:
I should've looked through. Thanks :)